pipert2/utils/publish_queue.py

```diff
@@ -7,33 +9,52 @@ class PublishQueue(object):
     """
 
     def __init__(self):
+        self._mp_queues = []
         self._queues = []
+        self.transmit = None
 
     def register(self, queue: Queue):
         """Register a new queue to publish to.
 
         """
 
-        self._queues.append(queue)
+        if isinstance(queue, Queue):
+            self._queues.append(queue)
+        else:
+            self._mp_queues.append(queue)
 
-    def put(self, value, block=False, timeout=1):
+    def put(self, message, block=False, timeout=1):
         """Publish a value to every registered queue.
 
         Args:
-            value: The value to push to every registered queue.
+            message: The value to push to every registered queue.
             block: Whether or not to block each queue when putting a message.
             timeout: How long to wait for each queue if block is true.
 
         """
 
         for q in self._queues:
-            if not block:
-                force_push_to_queue(q, value)
-            else:
-                try:
-                    q.put(value, block, timeout)
-                except Full as e:
-                    raise e
+            _push_to_queue(q, message, block, timeout)
+
+        if len(self._mp_queues) > 0:
+            transmitted_value = self.transmit(message.payload.data)
+            message.update_data(transmitted_value)
+            multiprocessing_message = Message.encode(message)
+
+            for q in self._mp_queues:
+                _push_to_queue(q, multiprocessing_message, block, timeout)
+
+
+def _push_to_queue(q, value, block, timeout):
+    if not block:
+        force_push_to_queue(q, value)
+    else:
+        try:
+            q.put(value, block, timeout)
+        except Full as e:
+            raise e
+
+    time.sleep(0.000001)
```

Comment:
Put it in a config file or a constants file with a specific name that explain why this you put this sleep